### PR TITLE
[MJAVADOC-810] [REGRESSION] MJAVADOC-791 causes forked Maven execution fail if any toolchains or settings file isn't present

### PR DESCRIPTION
**Pull request by cstamas**
As request carries the "effective" value (default or user overriden ones) but it does not mean they exist. And as invoker used, effect is same as user would use -s/-t CLI commands with non-existent files: mvn CLI failure.

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/javadoc/JavadocUtil.java [hunk 1]
```diff
@@ -781,9 +781,13 @@ protected static void invokeMaven(
         request.setBaseDirectory(projectFile.getParentFile());
         request.setPomFile(projectFile);
         request.setGlobalSettingsFile(globalSettingsFile);
-        request.setUserSettingsFile(userSettingsFile);
+        if (userSettingsFile.isFile()) {
```
Comment (cstamas) [inline src/main/java/org/apache/maven/plugins/javadoc/JavadocUtil.java:784]: more needed: UT will NPE as it tests with null value while in maven-core/embedder this value is never null.

**Comment by michael-o** [comment]
@cstamas I modified the code to respect the method contract and not to pass non-existing values.

**Comment by cstamas** [comment]
Lgtm

**Comment by michael-o** [comment]
Let's wait for IT and will create JIRA issue and merge.

**Comment by jira-importer** [comment]
Resolve #1187
